pkgs/by-name/nb/nbfc-linux/package.nix

```diff
@@ -0,0 +1,38 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  autoreconfHook,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nbfc-linux";
+  version = "0.1.15";
+
+  src = fetchFromGitHub {
+    owner = "nbfc-linux";
+    repo = "nbfc-linux";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-+xYr2uIxfMaMAaHGvvA+0WPZjwj3wVAc34e1DWsJLqE=";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+  ];
+  configureFlags = [
+    "--prefix=${placeholder "out"}"
+    "--sysconfdir=${placeholder "out"}/etc"
```

Comment:
Is this really right? I read your comment, but when i run `nbfc_service`, it says

```
nbfc_service: ERROR: /nix/store/rwbgycfp46n8fhywg3rynhb1fl6v2kh5-nbfc-linux-0.1.15/etc/nbfc/nbfc.json: No such file or directory
```

also `/nix/store/rwbgycfp46n8fhywg3rynhb1fl6v2kh5-nbfc-linux-0.1.15/etc/nbfc` is empty and I'd argue that it should read the config file in `/etc/nbfc` and not in the package.

Reply:
Yes having the conf in `/etc/nbfc/nbfc.json` would make more sense but im not sure how to achieve that

If it is set to`"--sysconfdir=/etc"`(such that conf will be located at /etc/nbfc/nbfc.json) the package wont build as it cannot make the directory /etc/nbfc 
<details>

```
❯ nix build
warning: Git tree '/data/dev/C/nbfc-linux' is dirty
error: builder for '/nix/store/dkdwwzgd6y2slqi7blmrfizb38fpcn4n-nbfc-linux.drv' failed with exit code 2;
       last 10 log lines:
       > sed -e 's|@BINDIR@|/nix/store/gi9j5j1vxwamdqbi34nmcrwgzfl3zfkm-nbfc-linux/bin|g' -e 's|@DATADIR@|/nix/store/gi9j5j1vxwamdqbi34nmcrwgzfl3zfkm-nbfc-linux/share|g' -e 's|@SYSCONFDIR@|/etc|g' -e 's|@RUNSTATEDIR@|/run|g' -e 's|@VERSION@|0.1.15|g' < completion/zsh/_nbfc_service.in > completion/zsh/_nbfc_service
       > installing
       > install flags: SHELL=/nix/store/226mxicx2n7hgdf2424vz5y526nlyly8-bash-5.2-p15/bin/bash install
       > install -Dm 755 src/nbfc_service  /nix/store/gi9j5j1vxwamdqbi34nmcrwgzfl3zfkm-nbfc-linux/bin/nbfc_service
       > install -Dm 755 src/ec_probe      /nix/store/gi9j5j1vxwamdqbi34nmcrwgzfl3zfkm-nbfc-linux/bin/ec_probe
       > install -Dm 755 src/nbfc          /nix/store/gi9j5j1vxwamdqbi34nmcrwgzfl3zfkm-nbfc-linux/bin/nbfc
       > # /usr/local/etc/nbfc
       > mkdir -p /etc/nbfc
       > mkdir: cannot create directory '/etc/nbfc': Permission denied
       > make: *** [Makefile:113: install-configs] Error 1
       For full logs, run 'nix log /nix/store/dkdwwzgd6y2slqi7blmrfizb38fpcn4n-nbfc-linux.drv'.
```

</details>

Explicitly setting `"--sysconfdir=${placeholder "out"}/etc"` is redundant as even without specifying it config dir is still $out/etc/, however it might be better to keep it in case the Makefile changes in the future


nbfc.json only specifies which preset to load (located in $out/share/nbfc/configs), by default its in /etc/nbfc/nbfc.json , however we can explicitly specify another location 

```
nbfc_service --config-file '/home/NixOS/.config/nbfc.json'"
```

and the contents of the json file would be something like 
```
{"SelectedConfigId": "Asus Zenbook Flip UX360UAK"}
```

Reply:
> Yes having the conf in `/etc/nbfc/nbfc.json` would make more sense but im not sure how to achieve that

Normally a package can not place anything in directories outside the nix store. To copy a file to `/etc` subdirs you need to use `environment.etc` NixOS option. You can add a NixOS module to do that!